storey/sources.py

```diff
@@ -730,212 +765,81 @@ def __init__(
-                                    if parsed_line is not None:
-                                        event = Event(parsed_line)
-                                    else:
-                                        event = Event(line)
-                                self.context.push_error(event, f"{ex}\n{message}", source=self.name)
-                            else:
-                                raise ex
-                if self._with_header:
-                    self._dates_indices = []
-
-        except BaseException as ex:
-            self._event_buffer.put(ex)
-        self._event_buffer.put(_termination_obj)
-
-    def _get_event(self):
-        event = self._event_buffer.get()
-        if isinstance(event, BaseException):
-            raise event
-        return event
+    def _get_element(self, body: OrderedDict):
+        if self._build_dict and self._with_header:
+            return dict(body)
+        return list(body.values())
 
-    async def _run_loop(self):
-        asyncio.get_running_loop().run_in_executor(None, self._blocking_io_loop)
-
-        def get_multiple():
-            events = [self._get_event()]
-            while not self._event_buffer.empty() and len(events) < 128:
-                events.append(self._get_event())
-            return events
+    def _get_by_field_or_index(self, field, body: OrderedDict, field_type: str, raise_exception=False):
+        result = None
+        if self._with_header and isinstance(field, str):
+            result = super()._get_by_field_or_index(
+                field=field, body=body, field_type=field_type, raise_exception=raise_exception
+            )
+        else:
+            result = list(body.items())[field][1]
+            if raise_exception:
+                self._is_nan_validator(result=result, body=body, field_type=field_type, field=field)
+        return result
 
-        while True:
-            events = await asyncio.get_running_loop().run_in_executor(None, get_multiple)
-            for event in events:
-                res = await self._do_downstream(event)
-                if event is _termination_obj:
-                    return res
+    def _field_validator(self, df, key_field, id_field, path="file path was not provided."):
+        path_message = get_path_message(path=path)
+        str_key_field = []
+        str_id_field = None
+        if id_field is not None:
+            if isinstance(id_field, str):
+                str_id_field = id_field
+            elif isinstance(id_field, int) and (id_field < 0 or id_field >= len(df.columns)):
+                raise IndexError(f"IndexError: id {id_field} is int and isn't in df index range.{path_message}")
+        if key_field:
+            key_field = [key_field] if not isinstance(key_field, list) else key_field
+            str_key_field = [key for key in key_field if isinstance(key, str)]
+            int_key_field = [key for key in key_field if isinstance(key, int)]
```

Comment:
This has gone way too far I think. We don't need to support a mix of ints and strings in `key_field`. It serves no useful purpose to the user.